test/telemetry_metrics_statsd_test.exs

```diff
@@ -118,16 +119,17 @@ defmodule TelemetryMetricsStatsdTest do
     start_reporter(
       metrics: [counter],
       port: port,
-      formatter: :datadog
+      formatter: :datadog,
+      default_tags: [env: "test"]
     )
 
     :telemetry.execute([:http, :request], %{latency: 172}, %{method: "GET", status: 200})
     :telemetry.execute([:http, :request], %{latency: 200}, %{method: "POST", status: 201})
     :telemetry.execute([:http, :request], %{latency: 198}, %{method: "GET", status: 404})
 
-    assert_reported(socket, "http.requests:1|c|#method:GET,status:200")
-    assert_reported(socket, "http.requests:1|c|#method:POST,status:201")
-    assert_reported(socket, "http.requests:1|c|#method:GET,status:404")
+    assert_reported(socket, "http.requests:1|c|#env:test,method:GET,status:200")
+    assert_reported(socket, "http.requests:1|c|#env:test,method:POST,status:201")
+    assert_reported(socket, "http.requests:1|c|#env:test,method:GET,status:404")
```

Comment:
Can we have one extra test which checks that metadata provided in `execute` overrides the global tags?